### PR TITLE
scanner: Encode paths to scan result files

### PR DESCRIPTION
**Pull request by haikoschol**
This change extracts `String.encodeOrUnknown()` from the downloader to utils and calls it in the scanner as well, when constructing the path to result files.

The issue this fixes came up when testing the toolkit with Go projects, where import paths are used as package names (e.g. `k8s.io/apiserver`). The scanner would then expect a directory `scanResults/k8s.io` to exist and fail to write the result file there.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/456)

### CODE REVIEW

File: downloader/src/main/kotlin/Main.kt [hunk 1]
```diff
@@ -35,13 +35,13 @@ import com.here.ort.model.MergedAnalyzerResult
 import com.here.ort.model.Package
 import com.here.ort.model.RemoteArtifact
 import com.here.ort.model.VcsInfo
+import com.here.ort.utils.encodeOrUnknown
```
Comment (mnonnenmacher) [inline downloader/src/main/kotlin/Main.kt]: Nit: This import should be further down, just where the import of fileSystemEncode was before.

File: utils/src/main/kotlin/Utils.kt [hunk 2]
```diff
@@ -394,6 +394,11 @@ fun String.fileSystemDecode(): String =
         // URLDecoder does decode "." and "*".
         java.net.URLDecoder.decode(this, "UTF-8")
 
+/**
+ * Return the string encoded for safe use as a file name or "unknown", if the string is empty.
+ */
+fun String.encodeOrUnknown(): String = this.fileSystemEncode().takeUnless { it.isBlank() } ?: "unknown"
```
Comment (mnonnenmacher) [inline utils/src/main/kotlin/Utils.kt]: Following the book this refactoring and the bug fix should be in separate commits. But I won't insist on it.

Comment (sschuberth) [reply utils/src/main/kotlin/Utils.kt]: And while at it, we should be able to drop the explicit return type and the `this.` from here.
